lib/src/simulator.dart

```diff
@@ -271,10 +274,12 @@ class Simulator {
     _simulationEndRequested = true;
   }
 
-  /// End Simulation and return the exception
-  static void throwException(Exception err, StackTrace stacktrace) {
-    _exception = err;
-    _stacktrace = stacktrace;
+  /// Collect [exception] and [stacktrace] and add to [_simExceptions] `List`.
```

Comment:
The `_simExceptions` list is private, and shouldn't be referenced in the public API docs.  Rather, describe the behavior of this function from the user's perspective.  Something like:
"Collects an [exception] and associated [stackTrace] triggered asynchronously during simulation to be thrown synchronously by [run].  Calling this function will end the simulation after this [tick] completes".